test/Main.purs

```diff
@@ -671,6 +671,44 @@ main = do
   parseTest "-6.0" (-6.0) number
   parseTest "+6.0" (6.0) number
 
+  -- test from issue #161
+  parseErrorTestMessage
+    (string " " <?> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string " " <|> string "-" <?> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string " " <?> "bad" <|> string "-" <?> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string " " <~?> \_ -> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string " " <|> string "-" <~?> \_ -> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string " " <~?> (\_ -> "bad") <|> string "-" <~?> \_ -> "failure")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    ("failure" <??> string " ")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    ("failure" <??> string " " <|> string "-")
+    "no"
+    "Expected failure"
+  parseErrorTestMessage
+    (string "x" <|> "failure" <??> string " " <|> "bad" <??> string "-")
+    "no"
+    "Expected failure"
```

Comment:
It might be helpful to have a `<$>` test in hear just in case.

Reply:
I added `<*` in there, too, and then got a little carried away.